packages/server-wallet/src/wallet/index.ts

```diff
@@ -99,8 +100,21 @@ export class Wallet implements WalletInterface {
   async joinChannel(_args: JoinChannelParams): Result {
     throw 'Unimplemented';
   }
-  async updateChannel(_args: UpdateChannelParams): Result {
-    throw 'Unimplemented';
+  async updateChannel({channelId, appData, allocations}: UpdateChannelParams): Result {
+    return Channel.transaction(async tx => {
+      const channel = await Store.getChannel(channelId, undefined);
+
+      const outcome = deserializeAllocations(allocations);
+
+      const decision = updateChannel({channelId, appData, outcome}, channel);
+      if (Either.isLeft(decision)) {
+        throw decision.left;
+      } else {
+        const {outgoing, channelResult} = await Store.signState(decision.right, tx);
+
+        return {outbox: outgoing.map(n => n.notice), channelResults: [channelResult]};
+      }
+    });
```

Comment:
Is there some clean way of encapsulating the bit that says "ensure this stuff happens in a single transaction on the 'channels' table" inside the `Store`? This is a similar point as before, where it seems like the locking mechanism of the database in question shouldn't need to be known to the `Wallet`, but could be an implementation detail by a `Store`.

Reply:
Yes, there is, and we can work on that next.